src/ripple/rpc/impl/TransactionSign.h

```diff
@@ -29,103 +30,38 @@ namespace RPC {
-class TxnSignApiFacade
-{
-private:
-    NetworkOPs* const netOPs_;
-    std::shared_ptr<ReadView const> ledger_;
-    AccountID accountID_;
-    std::shared_ptr<SLE const> sle_;
-
-public:
-    // Enum used to construct a Facade for unit tests.
-    enum NoNetworkOPs{
-        noNetOPs
-    };
-
-    TxnSignApiFacade () = delete;
-    TxnSignApiFacade (TxnSignApiFacade const&) = delete;
-    TxnSignApiFacade& operator= (TxnSignApiFacade const&) = delete;
-
-    // For use in non unit testing circumstances.
-    explicit TxnSignApiFacade (NetworkOPs& netOPs)
-    : netOPs_ (&netOPs)
-    { }
-
-    // For testTransactionRPC unit tests.
-    explicit TxnSignApiFacade (NoNetworkOPs noOPs)
-    : netOPs_ (nullptr) { }
-
-    // For testAutoFillFees unit tests.
-    TxnSignApiFacade (NoNetworkOPs noOPs, std::shared_ptr<ReadView const> ledger)
-    : netOPs_ (nullptr)
-    , ledger_ (ledger)
-    { }
-
-    void snapshotAccountState (AccountID const& accountID);
-
-    bool isValidAccount () const;
-
-    std::uint32_t getSeq () const;
-
-    boost::optional<STPathSet>
-    findPathsForOneIssuer (
-        AccountID const& dstAccountID,
-        Issue const& srcIssue,
-        STAmount const& dstAmount,
-        int searchLevel,
-        unsigned int const maxPaths,
-        STPathSet const& paths,
-        STPath& fullLiquidityPath) const;
-
-    void processTransaction (
-        Transaction::pointer& transaction,
-        bool bAdmin,
-        bool bLocal,
-        NetworkOPs::FailHard failType);
-
-    std::uint64_t scaleFeeBase (std::uint64_t fee) const;
-
-    std::uint64_t scaleFeeLoad (std::uint64_t fee, bool bAdmin) const;
-
-    bool hasAccountRoot () const;
-
-    error_code_i singleAcctMatchesPubKey (
-        RippleAddress const& publicKey) const;
-
-    error_code_i multiAcctMatchesPubKey (
-        AccountID const& acctID,
-        RippleAddress const& publicKey) const;
-
-    int getValidatedLedgerAge () const;
-
-    bool isLoadedCluster () const;
-};
-
 // A function to auto-fill fees.
-enum class AutoFill : unsigned char
-{
-    dont,
-    might
-};
-
 Json::Value checkFee (
     Json::Value& request,
-    TxnSignApiFacade& apiFacade,
     Role const role,
-    AutoFill const doAutoFill);
+    bool doAutoFill,
+    LoadFeeTrack const& feeTrack,
+    std::shared_ptr<ReadView const>& ledger);
 
-} // namespace detail
+// Return a std::function<> that calls NetworkOPs::processTransaction.
+using ProcessTransactionFn =
+    std::function<void (Transaction::pointer& transaction,
+        bool bAdmin, bool bLocal, NetworkOPs::FailHard failType)>;
+
+inline ProcessTransactionFn getProcessTxnFn (NetworkOPs& netOPs)
+{
+    return [&netOPs](Transaction::pointer& transaction,
+        bool bAdmin, bool bLocal, NetworkOPs::FailHard failType)
+    {
+        netOPs.processTransaction(transaction, bAdmin, bLocal, failType);
+    };
+}
 
+} // namespace detail
 
 /** Returns a Json::objectValue. */
 Json::Value transactionSign (
```

Comment:
Why have the extra overloads of `transactionSign`, can't we just pass the additional parameters at the call site? Where is the call site?